timer/config.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package timer
+
+import (
+	"time"
+)
+
+// UnlimitedBudget is a special value for the budget that means the timer has no budget
+var UnlimitedBudget = ^time.Duration(0) - 1
+
+// InheritedBudget is a special value for the budget that means the timer should inherit the budget from its parent
+// It is the default value if no options such as WithBudget, WithUnlimitedBudget or WithInheritedBudget are provided
+var InheritedBudget = time.Duration(0)
```

Comment:
Very optional... I wonder if its more clear that this is the 0-value of budget by making it:

```suggestion
var InheritedBudget time.Duration
```

Reply:
I forgot to put it to const. And const have to be assigned :(